src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala

```diff
@@ -164,7 +169,9 @@ object SparkExpressionToCExpression {
         Right(StringProducer.copyString(name))
       case Alias(AttributeReference(name, StringType, _, _), name2) =>
         Right(StringProducer.copyString(name))
-
+      case _ => sys.error(s"SparkExpressionToCExpression.evalString: " +
+        s"Could not match ${Try(expression.toString)}, " +
```

Comment:
In general, `toString` should not be `throw`ing, so `Try` may not be needed

Reply:
Right. In general. There are some expressions that throw an exception when you try to convert them into a string. I found out about them the hard way (LOL). Not sure what causes it. Recursive expression definitions? Not sure. But the "Try" is definitely needed in some situations.

Reply:
The `case _ => ... ` should actually be `case other => Left(other)`, I had missed this when writing the code.
The parent already handles the `Left`- side.

Reply:
Change made.